packages/expo-cli/src/commands/build-native/index.ts

```diff
@@ -72,7 +75,11 @@ export async function buildAction(projectDir: string, options: BuildOptions): Pr
     );
   }
   const easConfig: EasConfig = await new EasJsonReader(projectDir, platform).readAsync(profile);
-  const ctx = await createBuilderContextAsync(projectDir, easConfig);
+  const ctx = await createBuilderContextAsync(
+    projectDir,
+    easConfig,
+    options.parent?.nonInteractive === true
```

Comment:
```suggestion
    options.parent?.nonInteractive
```

Reply:
I think making it `boolean | undefined` instead of `boolean` will make it easy to forget pass this param to functions where it's used since then it'll become an optional param.

Reply:
I agree that context value should not be optional if possible, but you could move that check inside that function and maybe switch
`options.parent?.nonInteractive === true` -> `options.parent?.nonInteractive ?? false`
or
`options.parent?.nonInteractive === true` -> `!!options.parent?.nonInteractive`

Reply:
the same goes for credentials context

Reply:
I don't agree with @wkozyra95 :D I think `options.parent?.nonInteractive === true` is a lot cleaner.

Reply:
ok maybe it's just me, don't like that comparison ` === true` for some reason :D

@satya164 feel free to ignore that part, but I would still prefer to move that inside createBuilderContextAsync and Context constructor